manifests/service.pp

```diff
@@ -6,11 +6,23 @@
     fail("Use of private class ${name} by ${caller_module_name}")
   }
 
-  service { 'jenkins':
-    ensure     => $jenkins::service_ensure,
-    enable     => $jenkins::service_enable,
-    hasstatus  => true,
-    hasrestart => true,
+  if ($::operatingsystem != 'Amazon')
+  and (($::operatingsystem != 'Fedora' and versioncmp($::operatingsystemrelease, '7.0') >= 0)
+  or  ($::operatingsystem == 'Fedora' and versioncmp($::operatingsystemrelease, '15') >= 0)) {
```

Comment:
My preference would be to put the platform conditional logic in `params.pp` and set the provider param via a variable rather than duplicate the service resource.  Does that make sense?